docs/cloud/concepts/execution.md

```diff
@@ -0,0 +1,72 @@
+# Execution
+
+Executing your flows using Prefect Cloud is accomplished through two powerful abstractions—storage and environments. By combining Prefect's concepts of storage and environments, flows are able to be saved, shared, and executed across various supported platforms.
+
+## Storage
+
+[Storage](https://docs.prefect.io/api/unreleased/environments/storage.html) objects are pieces of functionality which define how and where a flow should be stored. Prefect currently has support for storage options ranging from ephemeral in-memory storage to Docker images which can be stored in registries.
+
+::: tip Cloud Acceptable Storage
+Currently the only supported Storage class in Prefect Cloud is [Docker storage](https://docs.prefect.io/api/unreleased/environments/storage.html#docker). This is due to the fact that Prefect Cloud does not retrieve the storage object itself and only cares about metadata describing the location of the image.
+:::
+
+### How Storage is Used
+
+To attach storage to your flows simply provide your storage object at initialization:
+
+```python
+from prefect.environments.storage import Docker
+
+f = Flow("example-storage", storage=Docker(registry_url="prefecthq/storage-example"))
+```
+
+or assign it directly:
+
+```python
+from prefect.environments.storage import Docker
+
+f = Flow("example-storage")
+f.storage = Docker(registry_url="prefecthq/storage-example")
+```
+
+When you deploy your flow to Prefect Cloud the storage object attached to the flow will be built. At this step the flow is serialized to byte code and placed inside of the storage. For added convenience, `flow.deploy` will accept arbitrary keyword arguments which will then be passed to the initialization method of your configured default storage class (which is `Docker` by default). Consequently, the following code will actually create a `Docker` object for you at deploy-time and push that image to your specified registry:
```

Comment:
I feel like it's worth calling out that users can optionally build the storage prior to deploy using `storage.build()` and then `build=False` in the `deploy` command.